src/hazard_manager/mod.rs

```diff
@@ -0,0 +1,439 @@
+#![allow(dead_code)]
+#![allow(unused_assignments)]
+#![allow(unused_variables)]
+use crate::common::{consts::*, *};
+use skyline::hooks::{getRegionAddress, Region};
+use skyline::hooks::A64InlineHook;
+use skyline::hook;
+use skyline::error::show_error;
+use smash::app::smashball::is_training_mode;
+use std::{iter::StepBy, ops::Range};
+
+struct TextIter<InnerIter: Iterator<Item = usize> + Sized> {
```

Comment:
Love this, but in the future we should add it to skyline-rs so it can be reused. Including the instruction defs, that way this file would be super clean. Not a blocker for this PR though.